### PR TITLE
(update) live messaging w/socket.io

### PR DESCRIPTION
**Pull request by mconner89**

### CODE REVIEW

**Review by gregthompson27 — APPROVED** [review]

File: server/index.js [hunk 1]
```diff
@@ -68,12 +68,23 @@ app.get('/*', (req, res) => {
 const server = http.createServer(app);
 const io = socketio(server);
 
-io.on('connection', (socket) => {
-  console.info('we have a new connection');
-  socket.on('create', (room) => {
+io.sockets.on('connection', (socket) => {
+  console.info('socket connected');
```
Comment (jstrzesz) [inline server/index.js:72]: remove

File: server/index.js [hunk 2]
```diff
@@ -68,12 +68,23 @@ app.get('/*', (req, res) => {
...
+  socket.on('room', (room) => {
     socket.join(room);
     console.info(`someone joined room ${room}`);
   });
+  socket.on('leave', (room) => {
+    socket.leave(room);
+    console.info(`someone left room ${room}`);
+  });
```
Comment (jstrzesz) [inline server/index.js:80]: remove

File: src/components/messages/MessageList.tsx [hunk 3]
```diff
@@ -1,30 +1,43 @@
...
   const [messages, setMessages] = useState([{ body: '', sender: true }]);
-  const classes = useStyles();
 
-  useEffect(() => {
+  const getMessages = async () => {
     const params = { thread, userId };
-    axios.get('message/getMessages', { params })
-      .then(({ data }) => setMessages(data));
+    const newMessages = await axios.get('message/getMessages', { params })
+      .then(({ data }) => data);
```
Comment (jstrzesz) [inline src/components/messages/MessageList.tsx:25]: .catch needed

**Review by jstrzesz — CHANGES_REQUESTED** [review]
see comments

**Review by slatzz — APPROVED** [review]

**Pull request merged**